design-proposals/vhostuser/vhostuser-interface.md

```diff
@@ -0,0 +1,143 @@
+
+# Overview
+- User Space networking is a feature that involves packet path from the VM to the data plane running in userspace like OVS-DPDK etc, by-passing the kernel.
+- With VhostUser interface, the kubevirt VM created can utilize the dpdk virtio functionality and avoid using kernel path for the traffic when DPDK is involved, allowing usage of fast datapath and improve performance.
+
+## Motivation
+- Any kubevirt VM uses linux kernel interfaces to connect to its host and to reach out externally or to communicate within the cluster. The current kubevirt VMs can only support the kernel data path for which the traffic needs to be through the kernel of the node/compute hosting the VM.
+
+- As a result, when we have a host with DPDK dataplane running in Userspace, the traffic from kubevirt VM will go through the kernel and reach the dataplane running in userspace. This longer path slows down the traffic between the application and the dataplane when both are running in userspace.
+
+- In order to improve the traffic performance and make use of the dpdk, we need to use a different path that can by-pass the kernel and reach the dataplane from the user space directly. This so called fast path will run within the user space and will allow fast packet processing.
+
+## Goals
+- The kubevirt VM when created with DPDK features, should be able to use fast datpath if the host supports the DPDK dataplane allowing fast packet processing to improve performance of the traffic.
+
+## Non Goals
+- The user space networking will require the host to a DPDK dataplane.
+- The kubevirtci creates kernel mode dataplane hosts for testing the VMs. It needs to be updated to support DPDK.
```

Comment:
This is listed under "Non Goals", so do you mean testing with DPDK is not a goal? There will be no tests?

Reply:
We have a test plan to bring up a VM with vHostUser interface and another interface type like Bridge.

Reply:
I don't understand why this point is listed under "Non Goals". Non goals are things that are out of scope. For example, supporting PCI network card device assignment would be a non goal because it's beyond what vhost-user-net is about.

Maybe this point about extending kubevirtci should go in the "Goals" section?